src/overlays/actors/ovl_En_Jso2/z_en_jso2.c

```diff
@@ -85,115 +223,1713 @@ ActorInit En_Jso2_InitVars = {
+            }
+            break;
+
+        case EN_JSO2_DEATH_CS_STATE_TELL_HINT:
+            if (EnJso2_ShouldAdvanceMessage(play)) {
+                if (this->actor.textId == 0x13AE) {
+                    this->actor.textId = 0x13AF;
+                } else if (this->actor.textId == 0x13AF) {
+                    this->actor.textId = 0x13B0;
+                    EnJso2_ChangeAnim(this, EN_JSO2_TAKE_OUT_BOMB);
+                } else if (this->actor.textId == 0x13B0) {
+                    play->msgCtx.msgLength = 0;
+                    if (this->bomb != NULL) {
+                        this->bomb->world.rot.z = 1;
+                        Player_SetCsActionWithHaltedActors(play, &this->actor, PLAYER_CSACTION_47);
+                        this->isFadingAway = true;
+                    }
+
+                    this->cutsceneTimer = 30;
+                    this->cutsceneState++;
+                    break;
+                }
+
+                Message_ContinueTextbox(play, this->actor.textId);
+            }
+            break;
+
+        case EN_JSO2_DEATH_CS_STATE_BLOW_UP_AND_FADE_AWAY:
+            Math_SmoothStepToS(&this->alpha, 0, 1, 15, 50);
+            Math_ApproachZeroF(&this->actor.shape.shadowScale, 0.3f, 3.0f);
+            if (this->cutsceneTimer == 0) {
+                this->actor.textId = 0x13B1;
+                Message_StartTextbox(play, this->actor.textId, &this->actor);
+                Actor_PlaySfx(&this->actor, NA_SE_EN_ANSATSUSYA_LAUGH);
+                this->timer = 50;
+                this->cutsceneState++;
+            }
+            break;
+
+        case EN_JSO2_DEATH_CS_STATE_WAIT_AND_END:
+            Player_SetCsActionWithHaltedActors(play, &this->actor, PLAYER_CSACTION_WAIT);
+            if (this->timer == 0) {
+                CutsceneManager_Stop(this->actor.csId);
+                Player_SetCsActionWithHaltedActors(play, &this->actor, PLAYER_CSACTION_END);
+                Actor_Kill(&this->actor);
+            }
+            break;
+
+        default:
+            break;
+    }
+
+    if (this->subCamId != SUB_CAM_ID_DONE) {
+        player->actor.world.pos.x = (Math_SinS(this->actor.world.rot.y) * 170.0f) + this->actor.world.pos.x;
+        player->actor.world.pos.z = (Math_CosS(this->actor.world.rot.y) * 170.0f) + this->actor.world.pos.z;
+        player->actor.world.rot.y = player->actor.shape.rot.y = this->actor.world.rot.y + 0x8000;
+        Matrix_RotateYS(((BREG(49) * 0x100) + this->actor.shape.rot.y + 0x1000), MTXMODE_NEW);
+        Matrix_MultVecZ(BREG(48) + 230.0f, &subCamEyeNextOffset);
+        this->subCamEyeNext.x = this->actor.world.pos.x + subCamEyeNextOffset.x;
+        this->subCamEyeNext.y = BREG(50) + -43.0f + this->actor.world.pos.y + 50.0f;
+        this->subCamEyeNext.z = this->actor.world.pos.z + subCamEyeNextOffset.z;
+        this->subCamAtNext.x =
+            player->actor.world.pos.x + ((this->actor.world.pos.x - player->actor.world.pos.x) * 0.5f);
```

Comment:
```suggestion
        this->subCamAtNext.x = F32_LERPIMP(player->actor.world.pos.x, this->actor.world.pos.x, 0.5f);
```
If you want, same for `.z`